packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -1149,8 +1158,11 @@ ${other.map((x) => `* ${x}`).join("\n")}
 
         const allArtifactsEmittedPerFile = solidityFilesCache.getEntries();
 
-        await artifacts.removeObsoleteArtifacts(allArtifactsEmittedPerFile);
-        await artifacts.removeObsoleteBuildInfos();
+        // We know this is the actual implementation, so we use some
```

Comment:
This is a bit ugly

Reply:
Yeah, maybe we should do the "function that receives Artifacts" thing.

Reply:
I don't remember what this means

Reply:
Instead of having `removeObsolete*` as instance methods of `Artifacts`, we convert them to functions that receive an `artifacts` instance.

Reply:
Oh, right. I'd say we leave it like this for now. It's internal, so we can change it whenever we want.